src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const { expenseSchema } = require('../schemas');
+const { dynamicSchema } = require('../utils');
+const { expensesService, usersService } = require('../services');
+
+const getAll = async (req, res) => {
+  const query = req.query;
+
+  const expenses = await expensesService.getAll(query);
+
+  res.json(expenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.json(expense);
+};
+
+const create = async (req, res) => {
+  const { error, value } = expenseSchema.validate(req.body, {
+    allowUnknown: true,
+  });
+
+  if (error) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const isUserExist = !!(await usersService.getById(value.userId));
+
+  if (!isUserExist) {
+    res.sendStatus(400);
```

Comment:
Instead of returning a generic 400 status code, consider sending a more specific error message indicating that the user does not exist, e.g., `res.status(404).json({ error: 'User not found' });`.